### DISCUSSION TITLE
problem in binary classification

### DISCUSSION THREAD

**Discussion by Sarthak-Singh04** [mrdbourke/pytorch-deep-learning#89] [Q&A] [answered]
I don't why they show error when I convert the numpy into tensor I follow the same as You used in the video pls help this line of code send me in trouble

```python 
# Turn data into tensors
# Otherwise this causes issues with computations later on
import torch
x = torch.from_numpy(X).type(torch.float)
y = torch.from_numpy(y).type(torch.float)

# View the first five samples
x[:5], y[:5]

TypeError                                 Traceback (most recent call last)
[<ipython-input-44-881c02e4b1fd>](https://localhost:8080/#) in <module>()
      2 # Otherwise this causes issues with computations later on
      3 import torch
----> 4 X = torch.from_numpy(X).type(torch.float)
      5 y = torch.from_numpy(y).type(torch.float)
      6 

TypeError: expected np.ndarray (got Tensor)
```

**Comment by aronvandepol** [comment, marked as answer]
You are passing a tensor object into a '.from_numpy()' function as it says in the video. The X and likely y variables contain a tensor. '.from_numpy()' needs a ndarray. As the error says. 

If your code is exactly the same it can be due to incorrect order of jupyter cells, or running it twice.

**Reply by Sarthak-Singh04** [reply]
okk bro I understand thanks for solving the problem

**Reply by Sarthak-Singh04** [reply]
l understand bro why  they show error in the code
step-1:X=torch.from_numpy(X).type(torch.float)
I run this code in the shell they work bcz our X is ndarray they convert the X into tensorthe X is stored in google colab memory location .
Step-2: If we re-run this shell again they show this error TypeError: expected np.ndarray (got Tensor) 
because in step-1 we convert the X into tensor  in the Google colab they store the X into tensor  we send again this tensor in the code i.e the reason they show error
